### PR TITLE
[MM-23003] Document sql store 'SqlTeamStore' methods (Task 8)  fixes

### PR DESCRIPTION
**Pull request by Ayanrocks**
#### Summary
This adds some comments for godoc to sqlTeamStore methods

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/14003

### CODE REVIEW

**Comment by jespino** [comment]
@Ayanrocks the tests failed because a problem with our CI, don't worry about it, it's not your fault :)

File: store/sqlstore/team_store.go [hunk 1]
```diff
@@ -1173,6 +1176,7 @@ func (s SqlTeamStore) GetTeamsForUserWithPagination(userId string, page, perPage
 	return dbMembers.ToModel(), nil
 }
 
+// GetChannelUnreadsForAllTeams returns all the messages that are unread in the channel for all teams.
```
Comment (jespino) [inline store/sqlstore/team_store.go:1179]: This is not accurate, it doesn't return the messages, it returns the information about unreads/mentions/notifyProps for all the channels in all the teams (except the excluded team).

File: store/sqlstore/team_store.go [hunk 2]
```diff
@@ -1153,6 +1153,9 @@ func (s SqlTeamStore) GetTeamsForUser(userId string) ([]*model.TeamMember, *mode
 	return dbMembers.ToModel(), nil
 }
 
+// GetTeamsForUserWithPagination returns limited TeamMembers according to the perPage parameter specified.
+// It also offsets the records as per the page parameter supplied.
+// Returns an error if an error occurs during querying.
```
Comment (jespino) [inline store/sqlstore/team_store.go:1158]: I think this line doesn't say much here, and we are not providing this text anywhere else, I would just omit this line. What do you think @justinegeffen?

File: store/sqlstore/team_store.go [hunk 3]
```diff
@@ -1194,6 +1198,7 @@ func (s SqlTeamStore) GetChannelUnreadsForAllTeams(excludeTeamId, userId string)
 	return data, nil
 }
 
+// GetChannelUnreadsForTeam returns all the unread messages in the channel for a single team
```
Comment (jespino) [inline store/sqlstore/team_store.go:1201]: Same problem here, we are returning the metadata (unreads/mentions/notifyProps) of all the channels in a team, but we are not returning the messages.

**Review by jespino — CHANGES_REQUESTED** [review]
Some changes required, but is looking good. Thanks @Ayanrocks. :tada:

File: store/sqlstore/team_store.go (hunk 3 above)
Comment (Ayanrocks) [reply store/sqlstore/team_store.go]: fixed it

File: store/sqlstore/team_store.go (hunk 1 above)
Comment (Ayanrocks) [reply store/sqlstore/team_store.go]: fixed it

**Comment by Ayanrocks** [comment]
Any updates on this? @jespino @justinegeffen

**Review by jespino — APPROVED** [review]
Now looks good to me, sorry the delay reviewing this.

File: store/sqlstore/team_store.go [hunk 4]
```diff
@@ -1173,6 +1175,8 @@ func (s SqlTeamStore) GetTeamsForUserWithPagination(userId string, page, perPage
 	return dbMembers.ToModel(), nil
 }
 
+// GetChannelUnreadsForAllTeams returns unreads msg count, mention counts and notifyProps,
```
Comment (justinegeffen) [inline store/sqlstore/team_store.go:1178]:
```suggestion
// GetChannelUnreadsForAllTeams returns unreads msg count, mention counts, and notifyProps
```

File: store/sqlstore/team_store.go [hunk 5]
```diff
@@ -1173,6 +1175,8 @@ func (s SqlTeamStore) GetTeamsForUserWithPagination(userId string, page, perPage
...
+// for all the channels in all the teams except the excluded ones
```
Comment (justinegeffen) [inline store/sqlstore/team_store.go:1179]:
```suggestion
// for all the channels in all the teams except the excluded ones.
```

File: store/sqlstore/team_store.go [hunk 6]
```diff
@@ -1194,6 +1198,7 @@ func (s SqlTeamStore) GetChannelUnreadsForAllTeams(excludeTeamId, userId string)
 	return data, nil
 }
 
+// GetChannelUnreadsForTeam returns unreads msg count, mention counts and notifyProps for all the channels in a single team
```
Comment (justinegeffen) [inline store/sqlstore/team_store.go:1201]:
```suggestion
// GetChannelUnreadsForTeam returns unreads msg count, mention counts, and notifyProps for all the channels in a single team.
```

File: store/sqlstore/team_store.go [hunk 7]
```diff
@@ -1375,6 +1380,7 @@ func (s SqlTeamStore) ResetAllTeamSchemes() *model.AppError {
 	return nil
 }
 
+// ClearCaches method not implemented
```
Comment (justinegeffen) [inline store/sqlstore/team_store.go:1383]:
```suggestion
// ClearCaches method not implemented.
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Minor punctuation suggestions but otherwise LGTM. Apologies for the delay!

**Comment by Ayanrocks** [comment]
Fixed all Requested changes. @justinegeffen

**Comment by Ayanrocks** [comment]
Any update on this. Its more than 5 business days. @justinegeffen @jespino

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Hi @Ayanrocks! I apologize for the delay, I thought I'd approved the PR when I submitted my suggestions. It looks good to me, thank you for the PR! :)

**Pull request merged**